exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,65 +669,179 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 	if err := s.Validate(e.GetTradingRequirements()); err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
+	oTypeString, err := orderTypeString(s.Type)
 	if err != nil {
 		return nil, err
 	}
-	return s.DeriveSubmitResponse(strconv.FormatInt(response.OrderNumber, 10))
+	switch s.AssetType {
+	case asset.Spot:
+		var smartOrder bool
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop:
+			smartOrder = true
+		}
+		if smartOrder {
```

Comment:
🚧 Change request:
Just realised this is a 2-step if condition.
I'll include in patch.
```suggestion
		switch s.Type {
		case order.Stop, order.StopLimit, order.TrailingStop:
```